File: src/01-easy/happy-numbers/solution.java
import java.util.*;
class Solution {
    static int dss(int n){int s=0;while(n>0){int d=n%10;s+=d*d;n/=10;}return s;}
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        while(n-->0){
            String s=sc.next();
            int x=0; for(char c:s.toCharArray()){int d=c-'0';x+=d*d;}
            Set<Integer> seen=new HashSet<>();
            while(x!=1&&seen.add(x)) x=dss(x);
            System.out.println(s+(x==1?" :)":" :("));
        }
    }
}

File: src/01-easy/happy-numbers/solution.vb
Module Solution
    Function Dss(n As Integer) As Integer
        Dim s As Integer = 0
        While n > 0
            Dim d As Integer = n Mod 10 : s += d*d : n \= 10
        End While
        Return s
    End Function
    Sub Main()
        Dim n As Integer = CInt(Console.ReadLine())
        For i As Integer = 1 To n
            Dim s As String = Console.ReadLine().Trim()
            Dim x As Integer = 0
            For Each c As Char In s
                Dim d As Integer = Asc(c) - 48 : x += d*d
            Next
            Dim seen As New HashSet(Of Integer)
            While x <> 1 AndAlso seen.Add(x)
                x = Dss(x)
            End While
            Console.WriteLine(s & If(x = 1, " :)", " :("))
        Next
    End Sub
End Module
